controller/server_controller.py

```diff
@@ -125,31 +123,45 @@ def cull_idle_jupyter_servers(body, name, namespace, logger, **kwargs):
     js_server_status = get_js_server_status(body)
     custom_resource_api = get_api(config.api_version, config.custom_resource_name)
     idle_seconds = int(body["status"].get("idleSeconds", 0))
+    now = pytz.UTC.localize(datetime.utcnow())
+    last_activity = now if js_server_status is None else js_server_status.get("last_activity", now)
+    try:
+        jupyter_server_started = datetime.fromisoformat(
+            body["metadata"]["creationTimestamp"].replace("Z", "+00:00")
+        )
+    except KeyError:
+        jupyter_server_started = now
+    jupyter_server_age_seconds = (now - jupyter_server_started).total_seconds()
+    last_activity_age_seconds = (now - last_activity).total_seconds()
     logger.info(
         f"Checking idle status of session {name}, "
         f"idle seconds: {idle_seconds}, "
         f"cpu usage: {cpu_usage}m, "
-        f"server status: {js_server_status}"
+        f"server status: {js_server_status}, "
+        f"age: {jupyter_server_age_seconds} seconds"
     )
-
-    now = pytz.UTC.localize(datetime.utcnow())
-    jupyter_server_is_idle = (
+    jupyter_server_is_idle_now = (
         cpu_usage <= config.CPU_USAGE_MILLICORES_IDLE_THRESHOLD
         and type(js_server_status) is dict
         and js_server_status.get("connections", 0) == 0
-        and js_server_status.get("kernels", 0) == 0
-        and (
-            now - js_server_status.get("last_activity", now).astimezone(pytz.UTC)
-        ).total_seconds()
-        > config.JUPYTER_SERVER_IDLE_CHECK_INTERVAL_SECONDS
+        and last_activity_age_seconds > config.JUPYTER_SERVER_IDLE_CHECK_INTERVAL_SECONDS
     )
-
-    if (
-        jupyter_server_is_idle
+    delete_idle_server = (
+        jupyter_server_is_idle_now
+        and idle_seconds_threshold > 0
         and idle_seconds + config.JUPYTER_SERVER_IDLE_CHECK_INTERVAL_SECONDS
```

Comment:
In real use-cases where `idle_seconds_threshold >> config.JUPYTER_SERVER_IDLE_CHECK_INTERVAL_SECONDS` this won't matter, but in cases where those two are comparable (ie testing?) this might actually lead to some unexpected behaviour, no? 
wouldn't `idle_seconds > idle_seconds_threshold` be more intuitive and safe?

Reply:
Yeah I will switch to `idle_seconds > idle_seconds_threshold`. In reality this results for the session getting shut down one idle check period after the threshold. Because the threshold is set by the user (or at least amalthea user) I wanted to be safe and cull exactly on the threshold. But changing this as you propose should not break the tests.